### PR TITLE
Automatic launch of the demo application

### PR DESCRIPTION
**Pull request by alspozdnyakov**

### CODE REVIEW

File: Behavioral.Automation.DemoBindings/Bootstrapper.cs [hunk 1]
```diff
@@ -12,6 +16,8 @@ public class Bootstrapper
         private readonly ITestRunner _runner;
         private readonly DemoTestServicesBuilder _servicesBuilder;
         private readonly BrowserRunner _browserRunner;
+        private static Process _coreRunProcess;
+        private const string BaseUrl = "http://localhost:4200";
```
Comment (SergeyRuzin) [inline Behavioral.Automation.DemoBindings/Bootstrapper.cs:20]: Can't we use the BaseUrl from ConfigService?

File: src/Behavioral.Automation/nuget.config [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <activePackageSource>
+        <add key="All" value="(Aggregate source)" />
+    </activePackageSource>
+    <packageSources>
+        <clear />
+        <add key="github" value="https://nuget.pkg.github.com/alspozdnyakov/index.json" />
+    </packageSources>
+    <packageSourceCredentials>
+        <github>
+            <add key="Username" value="alspozdnyakov" />
```
Comment (SergeyRuzin) [inline src/Behavioral.Automation/nuget.config:12]: @andrey-brovko Can we use such creds in open source lib?

File: Behavioral.Automation.DemoBindings/Bootstrapper.cs [hunk 3]
```diff
@@ -20,11 +26,61 @@ public Bootstrapper(IObjectContainer objectContainer, ITestRunner runner, Browse
             _browserRunner = browserRunner;
             _servicesBuilder = new DemoTestServicesBuilder(objectContainer, new TestServicesBuilder(_objectContainer));
         }
+        
+        private static void SetUpTestApp()
+        {
+            try
```
Comment (vladimir-s-quantory) [inline Behavioral.Automation.DemoBindings/Bootstrapper.cs:38]: `try ... catch` block is designed to catch exceptions, i.e. abnormal program behavior. Let's try and rewrite this

File: src/BlazorApp/Data/WeatherForecast.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace BlazorApp.Data;
+
+public class WeatherForecast
+{
+    public DateTime Date { get; set; }
+
+    public int TemperatureC { get; set; }
+
+    public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
```
Comment (vladimir-s-quantory) [inline src/BlazorApp/Data/WeatherForecast.cs:11]: I guess `Math.Round` is a better solution here :)

File: src/BlazorApp/Pages/Error.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace BlazorApp.Pages;
+
+[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+[IgnoreAntiforgeryToken]
+public class ErrorModel : PageModel
+{
+    public string? RequestId { get; set; }
+
+    public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
```
Comment (vladimir-s-quantory) [inline src/BlazorApp/Pages/Error.cshtml.cs:13]:
```suggestion
    public bool RequestIdCanBeShown => !string.IsNullOrEmpty(RequestId);
```

File: src/BlazorApp/Shared/MainLayout.razor [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+<PageTitle>Behavioral automation demo</PageTitle>
+
+<div class="page">
+    <div class="sidebar">
+        <NavMenu />
+    </div>
+
+    <main>
+        <div class="top-row px-4">
+            <a href="https://docs.microsoft.com/aspnet/" target="_blank">About</a>
```
Comment (vladimir-s-quantory) [inline src/BlazorApp/Shared/MainLayout.razor:12]: Do we need a link to MS docs?)

File: src/BlazorApp/Shared/NavMenu.razor [hunk 7]
```diff
@@ -0,0 +1,39 @@
+﻿<div class="top-row ps-3 navbar navbar-dark">
+    <div class="container-fluid">
+        <a class="navbar-brand" href="">BlazorApp</a>
```
Comment (vladimir-s-quantory) [inline src/BlazorApp/Shared/NavMenu.razor:3]: What is this link?

**Review by vladimir-s-quantory — CHANGES_REQUESTED** [review]

**Comment by andrey-brovko** [comment]
- Let's Rename _Tests folder to Tests?
- Let's move BlazorApp to the Test folder?

File: src/BlazorApp/Shared/NavMenu.razor [hunk 8]
```diff
@@ -1,6 +1,6 @@
 ﻿<div class="top-row ps-3 navbar navbar-dark">
     <div class="container-fluid">
-        <a class="navbar-brand" href="">BlazorApp</a>
+        <h1 class="navbar-brand">DemoApp</h1>
```
Comment (vladimir-s-quantory) [inline src/BlazorApp/Shared/NavMenu.razor:3]: Maybe smthn like `BDD Demo App`?

**Review by vladimir-s-quantory — APPROVED** [review]

**Review by SergeyRuzin — DISMISSED** [review]

**Review by ILastovka — APPROVED** [review]

File: src/Behavioral.Automation/Behavioral.Automation.csproj [hunk 9]
```diff
@@ -16,8 +16,9 @@ The whole automation code is divided into the following parts:
 - UI structure descriptive code
 - Supportive code</Description>
         <Copyright>Quantori Inc.</Copyright>
-        <PackageVersion>1.1.0</PackageVersion>
-        <RepositoryUrl>https://github.com/quantori/Behavioral.Automation</RepositoryUrl>
+        <PackageVersion>1.1.2</PackageVersion>
+        <RepositoryUrl>https://github.com/alspozdnyakov/Behavioral.Automation.git</RepositoryUrl>
```
Comment (andrey-brovko) [inline src/Behavioral.Automation/Behavioral.Automation.csproj:20]: The link is incorrect.

File: src/Behavioral.Automation/Behavioral.Automation.csproj [hunk 10]
```diff
@@ -40,5 +41,17 @@ The whole automation code is divided into the following parts:
       <PackageReference Include="Selenium.WebDriver" Version="4.1.0" />
       <PackageReference Include="SpecFlow" Version="3.9.40" />
     </ItemGroup>
+    <ItemGroup>
```
Comment (andrey-brovko) [inline src/Behavioral.Automation/Behavioral.Automation.csproj:44]: Remove please the obj folder.

File: Behavioral.Automation.DemoScenarios/Behavioral.Automation.DemoScenarios.csproj [hunk 11]
```diff
@@ -20,6 +20,10 @@
   <ItemGroup>
     <ProjectReference Include="..\Behavioral.Automation.DemoBindings\Behavioral.Automation.DemoBindings.csproj" />
   </ItemGroup>
+  
+  <ItemGroup>
+    <ProjectReference Include="..\src\BlazorApp\BlazorApp.csproj" />
```
Comment (andrey-brovko) [inline Behavioral.Automation.DemoScenarios/Behavioral.Automation.DemoScenarios.csproj:25]: Why do we need the ref?

File: Behavioral.Automation.DemoBindings/Bootstrapper.cs [hunk 12]
```diff
@@ -21,10 +26,58 @@ public Bootstrapper(IObjectContainer objectContainer, ITestRunner runner, Browse
...
+        private static void RunTestApp()
+        {
+            string testAppPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..", "..", "..", "..", "src",
+                "BlazorApp");
+
+            _coreRunProcess = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "cmd.exe",
```
Comment (andrey-brovko) [inline Behavioral.Automation.DemoBindings/Bootstrapper.cs:51]: Looks like this will work only for windows?

**Review by andrey-brovko — APPROVED** [review]

**Review by SergeyRuzin — APPROVED** [review]

**Pull request merged**